lib/matplotlib/backend_bases.py

```diff
@@ -2552,8 +2564,10 @@ def __init__(self, canvas, num):
         canvas.manager = self  # store a pointer to parent
         self.num = num
 
-        self.key_press_handler_id = self.canvas.mpl_connect('key_press_event',
-                                                            self.key_press)
+        if rcParams['toolbar'] != 'navigation':
+            self.key_press_handler_id = self.canvas.mpl_connect(
+                                                'key_press_event',
+                                                self.key_press)
```

Comment:
Should `self.key_press_handler_id` get defined with a `None` so that an `AttributeError` doesn't happen by accident?